### PR TITLE
Explain `draw_line` and `draw_rect` coordinates

### PR DESCRIPTION
**Pull request by dalexeev**
Add the following schemas and code examples.

---

It may not be obvious to many users (like me before) how exactly these methods work and that sometimes non-integer coordinates are needed.

Closes godotengine/godot#41023.

### CODE REVIEW

**Comment by dalexeev** [comment]
@clayjohn @kleonc Could you provide feedback on this? See also godotengine/godot#41239.

**Comment by mhilbrunner** [comment]
cc @clayjohn

File: tutorials/2d/custom_drawing_in_2d.rst [hunk 1]
```diff
@@ -135,9 +135,72 @@ call ``queue_redraw()`` from the ``_process()`` callback, like this:
...
     }
 
+Coordinates
+-----------
+
+The drawing API uses the CanvasItem coordinate system, not pixel coordinates.
+The CanvasItem's transform is applied. Also, you can use
+:ref:`draw_set_transform<class_CanvasItem_method_draw_set_transform>` or
+:ref:`draw_set_transform_matrix<class_CanvasItem_method_draw_set_transform_matrix>`
+to apply a custom transform additionally.
```
Comment (clayjohn) [inline tutorials/2d/custom_drawing_in_2d.rst:145]:
```suggestion
The drawing API uses the CanvasItem's coordinate system, not necessarily pixel
coordinates. Which means it uses the coordinate space created after applying
the CanvasItem's transform. Additionally, you can use
:ref:`draw_set_transform<class_CanvasItem_method_draw_set_transform>` or
:ref:`draw_set_transform_matrix<class_CanvasItem_method_draw_set_transform_matrix>`
to apply a custom transform.
```

File: tutorials/2d/custom_drawing_in_2d.rst [hunk 2]
```diff
@@ -135,9 +135,72 @@ call ``queue_redraw()`` from the ``_process()`` callback, like this:
...
+
+When using ``draw_line``, you should consider the width of the line.
+Lines with odd widths should be shifted by ``0.5``.
```
Comment (clayjohn) [inline tutorials/2d/custom_drawing_in_2d.rst:148]:
```suggestion
When using ``draw_line``, you should consider the width of the line.
When using a width that is an odd size, the position should be shifted
by ``0.5`` to keep the line centered as shown below.
```

**Review by clayjohn — COMMENTED** [review]
Great work. This is going to be very helpful

File: tutorials/2d/custom_drawing_in_2d.rst (hunk 1 above)
Comment (dalexeev) [reply tutorials/2d/custom_drawing_in_2d.rst]: Done.

File: tutorials/2d/custom_drawing_in_2d.rst (hunk 2 above)
Comment (dalexeev) [reply tutorials/2d/custom_drawing_in_2d.rst]: Done.

**Review by clayjohn — APPROVED** [review]
Looks good to me!

File: tutorials/2d/custom_drawing_in_2d.rst [hunk 3]
```diff
@@ -139,9 +139,74 @@ call ``queue_redraw()`` from the ``_process()`` callback, like this:
...
 
+Coordinates
+-----------
+
+The drawing API uses the CanvasItem's coordinate system, not necessarily pixel
+coordinates. Which means it uses the coordinate space created after applying
+the CanvasItem's transform. Additionally, you can use
+:ref:`draw_set_transform<class_CanvasItem_method_draw_set_transform>` or
+:ref:`draw_set_transform_matrix<class_CanvasItem_method_draw_set_transform_matrix>`
+to apply a custom transform.
```
Comment (kleonc) [inline tutorials/2d/custom_drawing_in_2d.rst:150]:
```suggestion
The drawing API uses the CanvasItem's coordinate system, not necessarily pixel
coordinates. Which means it uses the coordinate space created after applying
the CanvasItem's transform. Additionally, you can apply a custom transform on
top of it by using
:ref:`draw_set_transform<class_CanvasItem_method_draw_set_transform>` or
:ref:`draw_set_transform_matrix<class_CanvasItem_method_draw_set_transform_matrix>`.
```
(not sure if it should be formatted like I've suggested)

I'd reword it to make it more obvious that the additional custom transform is applied on top of the original transform (instead of e.g. replacing it).

cc @clayjohn to tell if it looks fine for you.

Comment (clayjohn) [reply tutorials/2d/custom_drawing_in_2d.rst:150]: Looks fine to me! I agree that it is more clear

**Review by kleonc — APPROVED** [review]

**Review by skyace65 — APPROVED** [review]
Looks good! Thanks!
